presto-main/src/test/java/com/facebook/presto/type/TestDecimalCasts.java

```diff
@@ -299,12 +299,20 @@ public void testDoubleToLongDecimalCasts()
         assertDecimalFunction("CAST(DOUBLE '-1234567890.0' AS DECIMAL(20,10))", decimal("-1234567890.0000000000"));
         assertDecimalFunction("CAST(DOUBLE '1234567890123456.9' AS DECIMAL(16,0))", decimal("1234567890123457"));
         assertDecimalFunction("CAST(DOUBLE '-1234567890123456.9' AS DECIMAL(16,0))", decimal("-1234567890123457"));
+        assertDecimalFunction("CAST(DOUBLE '1234567890123456789012345' AS DECIMAL(30,5))", decimal("1234567890123456800000000.00000"));
```

Comment:
Is this a valid double value? Does it fit? You could try using hex floating point literals to represent test doubles: https://stackoverflow.com/questions/25712348/hexadecimal-floating-point-literals.

According to standard we want to make sure that Mantissa is not truncated right? What is the implicit precision of double?

Reply:
After offline discussion. It would be nice to have round trip tests to ensure mantissa information is not truncated.

Reply:
It is string which can be interpreted as a valid double (this one: 1234567890123456824475648)